app/main.py

```diff
@@ -1,6 +1,29 @@
-from typing import Callable
+from functools import wraps
+from typing import Callable, Tuple, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    func_cache = {}
+
+    @wraps(func)
+    def wrapper(*args: Any) -> Any:
+        if args in func_cache:
```

Comment:
Using mutable types like lists or dictionaries as keys in a dictionary will raise a TypeError because they are not hashable. Ensure that the arguments used as keys in `func_cache` are immutable. Consider converting mutable arguments to immutable types, such as tuples, before using them as keys.